sdk/monitor/Azure.Monitor.Query/src/QueryTimeSpan.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+
+namespace Azure.Monitor.Query
+{
+    /// <summary>
+    /// Represents a span of time over which the query would be executed.
+    /// </summary>
+    public readonly struct QueryTimeSpan : IEquatable<QueryTimeSpan>
+    {
+        /// <summary>
+        /// Represents the maximum <see cref="QueryTimeSpan"/>.
+        /// </summary>
+        public static QueryTimeSpan MaxValue => new QueryTimeSpan(TimeSpan.MaxValue);
+
+        /// <summary>
+        /// Gets the duration of the interval.
+        /// </summary>
+        public TimeSpan Duration { get; }
+
+        /// <summary>
+        /// Gets the start time of the interval.
+        /// </summary>
+        public DateTimeOffset? StartTime { get; }
+
+        /// <summary>
+        /// Gets the end time of the interval.
+        /// </summary>
+        public DateTimeOffset? EndTime { get; }
+
+        /// <summary>
+        /// Initializes an instance of <see cref="QueryTimeSpan"/> using a duration value.
+        /// </summary>
+        /// <param name="duration">The duration of the interval.</param>
+        public QueryTimeSpan(TimeSpan duration)
```

Comment:
what is the difference between these calling this with a duration of 1d, and calling `QueryTimeSpan(DateTimeOffset.Now.AddDays(-1), DateTimeOffset.Now)` ?

Reply:
client time vs service time. When only the duration is specified the time span is calculated during the query execution. In your example, it's pre-calculated on the client.

Reply:
You just re-wrote the comment for this method 👨‍🏫 😄

Reply:
![https://media4.giphy.com/media/rApKuVWCZZgvm/200.gif](https://media4.giphy.com/media/rApKuVWCZZgvm/200.gif)